tools/generate_from_protos.sh

```diff
@@ -70,4 +67,52 @@ for plugin in ${plugin_list}; do
     ${protoc_binary} -I ${proto_dir} --custom_out=${tmp_output_dir} --plugin=protoc-gen-custom=${protoc_gen_dcsdk} --custom_opt="file_ext=h,template_path=${template_path_mavsdk_server}" ${proto_dir}/${plugin}/${plugin}.proto
     mkdir -p ${script_dir}/../src/backend/src/plugins/${plugin}
     mv ${tmp_output_dir}/${plugin}/$(snake_case_to_camel_case ${plugin}).h ${script_dir}/../src/backend/src/plugins/${plugin}/${plugin}_service_impl.h
+
+    file_impl_h="${script_dir}/../src/plugins/${plugin}/${plugin}_impl.h"
+    if [[ ! -f "${file_impl_h}" ]]; then
+        ${protoc_binary} -I ${proto_dir} --custom_out=${tmp_output_dir} --plugin=protoc-gen-custom=${protoc_gen_dcsdk} --custom_opt="file_ext=h,template_path=${template_path_plugin_impl_h}" ${proto_dir}/${plugin}/${plugin}.proto
+        mv ${tmp_output_dir}/${plugin}/$(snake_case_to_camel_case ${plugin}).h ${file_impl_h}
+        echo "-> Creating ${file_impl_h}"
+    else
+        # Warn if file is not checked in yet.
+        if [[ ! $(git ls-files --error-unmatch ${file_impl_h} 2> /dev/null) ]]; then
+            echo "-> Not creating ${file_impl_h} because it already exists"
+        fi
```

Comment:
Personal preference: I think this adds noise to the script, it will mostly never run (hence can easily get outdated) and if it happens while we are working on templates, then it will spam us every time we run the script :grin:.

This said, I could live with it :wink:.

Reply:
No, it shouldn't add noise. The output only appears if the files are there but have not been committed. Without this it's confusing if you change the proto but the changes are not reflected in the generated impl. That's because I don't want to overwrite any changed that were made.